site/src/components/Resources/ResourceAvatar.tsx

```diff
@@ -22,15 +24,13 @@ const iconByResource: Record<string, typeof MemoryIcon> = {
   google_compute_instance: AdjustedMemoryIcon,
   aws_instance: AdjustedMemoryIcon,
   kubernetes_deployment: AdjustedMemoryIcon,
-  null_resource: HelpIcon,
+  null_resource: WidgetsIcon,
 }
 
-export type ResourceAvatarProps = { type: string }
+export type ResourceAvatarProps = { type: ValidIcons }
```

Comment:
`string` is too wide. By creating `ValidIcons`, we have a string union that accurately reflects `type`.

Reply:
looks like a test is now failing?

Reply:
> Type 'string' is not assignable to type 'ValidIcons'.

Ahh...I think the type in the component props needs to be updated.

Reply:
Okay, so it turns out there is a mismatch between types generated by backend and expectations on frontend. Backend says it can be of type `string` but really, it is a limited set of strings for which we have Icons.

Moving to draft so I can fix that first.

Reply:
I decided that will take more work and should be done as a separate PR so I'll clean this up and open a separate issue to address the types mismatch.

Reply:
https://github.com/coder/coder/issues/3767